drivers/docker/network.go

```diff
@@ -28,17 +28,24 @@ func (d *Driver) CreateNetwork(allocID string, createSpec *drivers.NetworkCreate
 		return nil, false, fmt.Errorf("failed to connect to docker daemon: %s", err)
 	}
 
-	repo, _ := parseDockerImage(d.config.InfraImage)
-	authOptions, err := firstValidAuth(repo, []authBackend{
-		authFromDockerConfig(d.config.Auth.Config),
-		authFromHelper(d.config.Auth.Helper),
-	})
-	if err != nil {
-		d.logger.Debug("auth failed for infra container image pull", "image", d.config.InfraImage, "error", err)
-	}
-	_, err = d.coordinator.PullImage(d.config.InfraImage, authOptions, allocID, noopLogEventFn, d.config.infraImagePullTimeoutDuration, d.config.pullActivityTimeoutDuration)
-	if err != nil {
-		return nil, false, err
+	repo, tag := parseDockerImage(d.config.InfraImage)
+
+	// Pull the InfraImage only if it doesn't exist locally, or uses the "latest" tag
+	if dockerImage, _ := client.InspectImage(d.config.InfraImage); dockerImage == nil || tag == "latest" {
+		authOptions, err := firstValidAuth(repo, []authBackend{
+			authFromDockerConfig(d.config.Auth.Config),
+			authFromHelper(d.config.Auth.Helper),
+		})
+		if err != nil {
+			d.logger.Debug("auth failed for infra container image pull", "image", d.config.InfraImage, "error", err)
+		}
+		_, err = d.coordinator.PullImage(d.config.InfraImage, authOptions, allocID, noopLogEventFn, d.config.infraImagePullTimeoutDuration, d.config.pullActivityTimeoutDuration)
+		if err != nil {
+			return nil, false, err
+		}
+	} else {
+		// Image exists, so just increment its reference count
+		d.coordinator.IncrementImageReference(dockerImage.ID, d.config.InfraImage, allocID)
```

Comment:
Review question: Is it appropriate to invoke `IncrementImageReference` here?  I adapted this logic from `createImage`, but I don't know if there is maybe a reason to not use `IncrementImageReference` in this context. https://github.com/hashicorp/nomad/blob/9c582eaf6c0a68c1e32be45057161c8a0405b0bf/drivers/docker/driver.go#L537-L545

Also, for the `callerID` string (3rd argument) I used the `allocID`, because there's no task ID available like in `driver.go`.  I'm not sure if that's the most appropriate value though?